bm-persona/Libraries/LibraryDataSource/Library.swift

```diff
@@ -52,6 +53,27 @@ class Library: SearchItem {
         self.imageURL = URL(string: imageLink ?? "")
     }
     
+    var distanceToUser: Double = Double.nan
+    
+    func findDistanceToUser(userLoc: CLLocation) -> Double {
+        if  let libLat = self.latitude,
+            let libLong = self.longitude,
+            !libLat.isNaN && !libLong.isNaN {
+            let libLoc = CLLocation(latitude: libLat, longitude: libLong)
+            let distance = round(userLoc.distance(from: libLoc) / 1600.0 * 10) / 10
+            distanceToUser = distance
+            return distanceToUser
+        }
+        return Double.nan
+    }
+    
+    func getDistanceToUser(userLoc: CLLocation) -> Double {
+        if distanceToUser.isNaN {
+            return findDistanceToUser(userLoc: userLoc)
+        }
+        return distanceToUser
+    }
+    
```

Comment:
Can we move this to another class that resources like Libraries subclass, since we'll likely be using it for things other than libraries?

Reply:
Moved to protocol HasLocation